### DISCUSSION TITLE
关于Clone接口的使用

### DISCUSSION THREAD

**Discussion by nichujie** [joint-online-judge/horse#70] [Q&A]
![image](https://user-images.githubusercontent.com/17191827/143619329-94237bcd-fbf7-449e-b782-e4c28898348e.png)

目前正常的流程是：
* 在domain里创建problem
* 创建problem set
* 将problem加入problem set

对于上图中的clone接口，既然有`problemSet`这个字段，是否意味着clone problem的逻辑其实包含了2步：
* 先将domain A中的problem P克隆到domain B
* 再将克隆后的problem P‘加入到domain B的problem set中

这样带来的问题就有：
* 操作分两步，现在是否加了transaction？（如果是添加多个problem也需要transaction保证）
* 如果我只想克隆problem到我的domain，但并不想加入problem set怎么办？

建议解决方案：
* 添加transaction
* problemSet作为可选字段

**Comment by nichujie** [comment]
@tc-imba @BoYanZh

**Reply by BoYanZh** [reply]
transaction在加了（指写到TODO
